### ISSUE TITLE
Port the tests to stable

### ISSUE THREAD

**Issue by antoyo** [antoyo/tql#10]
By copy-pasting the files in [this crate](https://github.com/antoyo/tql/tree/master/testcrate) and creating a `compilefail` test instead of a `ui` test.

**Comment by Atul9** [comment]
@antoyo I would like to take this up. I have a question - copy-pasting from https://github.com/antoyo/tql/tree/master/tests/testcrate to where? Did you mean to create stablecrate in tql/test? & rename ui to compilefail folder?

**Comment by antoyo** [comment]
Instead of copy-pasting, I'd actually prefer to use the `include!` macro.
Also, it would be nice to add a test that will check if we're missing some tests (to ease maintenance).
So, yes, we can start with the `testcrate` tests, so create a new crate similar to `testcrate`.
Since on stable we don't have the right position in the error message, it will be a `compilefail` (name to be checked) test instead of `ui`, so you'll need to update [this line](https://github.com/antoyo/tql/blob/master/tests/testcrate/tests/fail.rs#L46) in the new crate.
Moreover, you won't need the `.stderr` files (because they belong to `ui` tests), but you'll need to update the error comments [like this one](https://github.com/antoyo/tql/blob/master/tests/testcrate/tests/ui/aggregate.rs#L50) in this `testcrate` crate because the new crate will only contains `include!()` macros.
Thanks for your help and if you need more info, feel free to ask.

To sum up:

 * Create a new crate `compilefail-stable` (or whatever name) in the `tests` folder.
 * Copy-paste the `fail.rs` from `compilefail` tests and change the mode to `compilefail` (or whatever the name is).
 * Create one with per compile fail test where each file contains an `include!()` macro call.
 * Add a normal Rust test (alongside `fail`) which will check that all the files in `testcrate/tests/ui` are in `testcrate-stable/tests/fail`.
 * Add the new command to launch this new test on stable in the travis file.

I just thought that using `include!()` might create issues with `compiletest`, so maybe we won't be able to use it (please test it).
